### DISCUSSION TITLE
Optax vs Cvxpy

### DISCUSSION THREAD

**Discussion by pushkar5586** [google-deepmind/optax#881] [General] [answered]
Hi All,

Given that one can solve almost any convex optimization problem by writing down the objective function in Jax, and optimizing it using an appropriate gradient descent algorithm in Optax; is there any inherent benefit of using Cvxpy over Jax+Optax?

Thanks.

**Comment by fabianp** [comment, marked as answer]
cvxpy comes with convergence guarantees, which, at the moment, optax lacks. Also, currently most of the solvers in optax require some hyperparameter tuning (like the learning rate). I think most of the solvers in cvxpy on the other hand, don't require hyperparameter tuning.

The downside of cvxpy is that it's not applicable to nonconvex problems, and that if often scales worse than optax for very large problems

**Reply by pushkar5586** [reply]
Thanks. Your comments about hyper parameter tuning and convergence are also inline with my experience.
